js/compressed-token/src/token-serde.ts

```diff
@@ -0,0 +1,150 @@
+import { Connection, PublicKey } from '@solana/web3.js';
+import {
+  Utxo_IdlType,
+  getMockRpc,
+  PublicTransactionEvent_IdlType,
+  addMerkleContextToUtxo,
+  MerkleContext,
+  defaultTestStateTreeAccounts,
+  createUtxoHash,
+} from '@lightprotocol/stateless.js';
+import { CompressedTokenProgram } from './program';
+import { TokenTlvData_IdlType } from './types';
+import { WasmFactory } from '@lightprotocol/account.rs';
+import { BN } from '@coral-xyz/anchor';
+
+// TODO: later consider to implement a coherent struct (test with ix creation)
+export type UtxoWithParsedTokenTlvData = {
+  utxo: Utxo_IdlType;
+  parsed: TokenTlvData_IdlType;
+  merkleContext: MerkleContext | null;
+};
+
+export type EventWithParsedTokenTlvData = {
+  inUtxos: UtxoWithParsedTokenTlvData[];
+  outUtxos: UtxoWithParsedTokenTlvData[];
+};
+
+/** @internal */
+export function parseTokenLayoutWithIdl(
+  utxo: Utxo_IdlType,
+): TokenTlvData_IdlType | null {
+  if (utxo.data === null) {
+    return null;
+  }
+
+  if (utxo.data.tlvElements.length === 0) return null;
+  if (
+    /// We can assume 0th element is token
```

Comment:
Is this a good assumption?
What if we have custom pdas with different data, then the process crashes right?